src/main/java/com/google/step/snippet/data/Card.java

```diff
@@ -5,12 +5,17 @@
   private final String code;
   private final String link;
   private final String description;
+  private final long upvote;
+  private final long downvote;
 
-  public Card(String title, String code, String link, String description) {
+  public Card(
+      String title, String code, String link, String description, long upvote, long downvote) {
     this.title = title;
     this.code = code;
     this.link = link;
     this.description = description;
+    this.upvote = upvote;
```

Comment:
I recall you mentioning previously that both upvotes and downvotes (these should probably be plural? singular makes it sound like a boolean) are kept in the backend for future unknown use. But the frontend only cares about the difference. Let's only give the frontend exactly as much information as they need, so only the difference?

Reply:
This has ramifications on a lot of your PR - for example, Client.java only needs one method that tells you what the feedback value (difference) is, not a method to get back an entity (which is an implementation detail), one for upvotes, and one for downvotes.